pixano/app/api/items.py

```diff
@@ -411,107 +411,118 @@ def find_top_parent_entity(entity: Entity) -> Entity | None:
-                            bboxes.append(
-                                {
-                                    "coords": annotation.xywh_coords,
-                                    "format": "xywh",
-                                    "is_normalised": annotation.is_normalized,
-                                    "confidence": annotation.confidence,
-                                    "view_id": annotation.view_ref.name,  # danger faux-ami !
-                                    "frame_index": frame_index,
-                                    "is_key": (annotation.is_key if hasattr(annotation, "is_key") else True),
-                                    "is_thumbnail": False,
-                                    "tracklet_id": annotation.entity_ref.id,
-                                }
-                            )
-                        if is_keypoints(type(annotation), False) and annotation != NoneKeypoints:
-                            # features.update(getFeatures(annotation, KeyPoints))
-                            keypoints.append(
-                                {
-                                    "template_id": annotation.template_id,
-                                    "vertices": annotation.map_back2front_vertices(),
-                                    "frame_index": frame_index,
-                                    "view_id": annotation.view_ref.name,
-                                    "is_key": (annotation.is_key if hasattr(annotation, "is_key") else True),
-                                    "tracklet_id": annotation.entity_ref.id,
-                                }
-                            )
+                    trackid2entityidlist[entity.parent_ref.id].append(entity.id)
+                    entity_id2trackid[entity.id] = entity.parent_ref.id
+
+        # gather annotations/track references
+        for annotation_group in groups[_SchemaGroup.ANNOTATION]:
+            for annotation in getattr(item, annotation_group):
+                if is_tracklet(type(annotation)):
+                    continue
+                entity_id = annotation.entity_ref.id
+                track_id = entity_id2trackid[entity_id]
+
+        # Load all tracks related data
+        track_bboxes = defaultdict(list)
+        track_keypoints = defaultdict(list)
+        track_features: dict[str, dict] = defaultdict(dict)
+        kept_track_ids = []
+        for annotation_group in groups[_SchemaGroup.ANNOTATION]:
+            for annotation in getattr(item, annotation_group):
+                if is_tracklet(type(annotation)):
+                    continue
+                entity_id = annotation.entity_ref.id
+                track_id = entity_id2trackid[entity_id]
+                frame_index = next(
+                    (
+                        view["frame_index"]
+                        for view in views[annotation.view_ref.name]
+                        if view["id"] == annotation.view_ref.id
+                    ),
```

Comment:
we suppose that the view is sequene frame which is ok for now but we should keep in mind that later we will support videos.